src/_pytest/tmpdir.py

```diff
@@ -45,8 +45,21 @@ def from_config(cls, config) -> "TempPathFactory":
             given_basetemp=config.option.basetemp, trace=config.trace.get("tmpdir")
         )
 
+    def _check_basename(self, basename: str):
+        basename = os.path.normpath(basename)
+        if (
+            os.path.isabs(basename)
```

Comment:
I believe we only need the last check (`(self.getbasetemp() / basename).resolve().parent != self.getbasetemp()`), can you check if the test still passes just with it? If so I think we should remove the first two checks as they are not necessary.